src/main/kotlin/com/sourcegraph/cody/context/ui/AddRepositoryDialog.kt

```diff
@@ -61,11 +61,10 @@ class AddRepositoryDialog(
               val codebaseName =
                   runCatching { convertGitCloneURLToCodebaseNameOrError(text) }.getOrNull()
               codebaseName ?: return@custom false
-              val repo =
-                  RemoteRepoUtils.getRepository(project, codebaseName)
-                      .completeOnTimeout(null, 2, TimeUnit.SECONDS)
-                      .get()
-              repo != null
+              RemoteRepoUtils.getRepositories(project, listOf(codebaseName))
+                  .completeOnTimeout(null, 2, TimeUnit.SECONDS)
+                  .get()
+                  .isNotEmpty()
```

Comment:
nit: I guess it should be a singleton, right?

Reply:
Changes